src/main/remote-sync/RemoteSyncManager.ts

```diff
@@ -0,0 +1,347 @@
+import Logger from 'electron-log';
+import * as helpers from './helpers';
+import {
+  RemoteSyncStatus,
+  RemoteSyncedFolder,
+  RemoteSyncedFile,
+  SyncConfig,
+  SYNC_OFFSET_MS,
+} from './helpers';
+import { reportError } from '../bug-report/service';
+import { obtainToken } from '../auth/service';
+import { DatabaseAdapter } from '../database/adapters/base';
+
+export class RemoteSyncManager {
+  private foldersSyncStatus: RemoteSyncStatus = 'IDLE';
+  private filesSyncStatus: RemoteSyncStatus = 'IDLE';
+  private status: RemoteSyncStatus = 'IDLE';
+  private onStatusChangeCallback: (newStatus: RemoteSyncStatus) => void =
+    () => {
+      return;
+    };
+  constructor(
+    private db: {
```

Comment:
Why don't limit the RemoteSyncManager to one database? on the methods, we are doing the same for files and folders. We could extract the `syncRemoteFiles` and `syncRemoteFolders` this we are not coupled to the data fetching. It will be easier to test since will be no need to mock the global fetch and will need less than half of the test since there will be no combinations to test (I'm thinking of cases where `syncFiles` and `syncFolders` are different of `true`)

Reply:
but this is coupled to the data fetching, is on purpose, that's why is called RemoteSyncManager, cause it syncs from remote, if we are going to sync from another source, we would need another system don't you think? I understand the idea but if we need to place stuff in the local db in another way, you can simply use the DriveFilesDB or the DriveFoldersDB

Reply:
The goal of extracting the data fetching is to avoid having to redo things and improve testability. As the file fetching is already implemented on the remote filesystem bus as it's coupled to the file system you need to implement it again on the manager and I'm having to implement it for Webdav. Plus being coupled makes it harder to extend the functionality, and keep it maintained

Reply:
Ok, so for example the RemoteSyncManager should receive maybe an input of `RemoteSyncedFile[]` and `RemoteSyncedFolder[]` but without containing the logic to retrieve those inputs?